pageserver/src/tenant/tasks.rs

```diff
@@ -103,6 +103,11 @@ async fn compaction_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
                 }
             }
 
+            if cancel.is_cancelled() {
```

Comment:
This is now a bit racy. Adding the error is good, and we should filter against such values knstead when logging near lines L114R119.

Reply:
nevermind I missed we turn this into an Ok at the outermost fn. This check is then just extra lines, correct?

Also, still too many `info!`, zero are needed when shutting down but this PR at least gets rid of one extra stacktrace.

Reply:
I tend to agree about the `info!` being a bit too verbose, I put it in here for consistency with the other cancel checks in the function.

The functional impact of this check is just to avoid calling into compaction_iteration (which emits an error log when state isn't active), rather than calling in there, logging error, returning and then dropping out of this loop further down when we next enter a `timeout`.